src/client/dfuse/pil4dfs/int_dfs.c

```diff
@@ -110,6 +111,48 @@ static uint16_t               eq_count_max;
 static uint16_t               eq_count;
 static uint16_t               eq_idx;
 
+/* the number of EQs for aio contexts. (aio_eq_count + eq_count) <= eq_count_max */
+static uint16_t               aio_eq_count;
+
+struct aio_eq {
+	daos_handle_t    eq;
+	pthread_mutex_t  lock;
+	_Atomic uint64_t n_op_queued;
+	_Atomic uint64_t n_op_done;
+};
+
+/* list of EQs dedicated for aio contexts. */
+static struct aio_eq aio_eq_list[MAX_EQ];
+/* the accumulated iodepth for aio context. */
+static long int      depth_sum_accu;
```

Comment:
I do not understand what you mean here by accumulated IO depth.
and if this is on an aio context, why is it global? i think the comment is wrong

Reply:
depth_sum_accu is increased by iodepth when a new aio context is created. Number of "iodepth" event queues (EQs) are created if the total number of EQs is not beyond the set limit.
It is used to determine "first_eq".
aio_ctx->first_eq = depth_sum_accu % aio_eq_count_g;
This is introduced to distribute IO requests from aio context evenly over available EQs and minimize the chances EQs are used by multiple aio contexts.

Reply:
OK i had to read the code again to understand what you mean.
if this is the running total of io depths from all io contexts, then please use a clearer name:
io_depth_total_g ? or something to that end..

also update the comment please on why this is needed (choosing the first eq for every io context request submission)